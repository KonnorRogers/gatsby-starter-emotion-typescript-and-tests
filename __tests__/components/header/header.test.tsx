import React from "react";
import { render } from "@testing-library/react";
import Header from "~components/header";

describe("Header component", () => {
  test("Should render a header with the given testid", () => {
    const { asFragment, getByTestId } = render(
      <Header siteTitle="test-title" className="header" data-testid="header" />
    );

    const header = getByTestId("header");
    expect(header).toHaveClass("header");
    expect(header).toHaveTextContent("test-title");
    expect(asFragment()).toMatchSnapshot();
  });
});
