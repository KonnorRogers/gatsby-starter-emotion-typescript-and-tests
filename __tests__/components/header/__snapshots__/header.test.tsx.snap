// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing Renders a header without a siteTitle defined 1`] = `
<DocumentFragment>
  .emotion-3 {
  background: rebeccapurple;
  margin-bottom: 1.45rem;
}

.emotion-2 {
  margin: 0 auto;
  max-width: 960px;
  padding: 1.45rem 1.1rem;
}

.emotion-1 {
  margin: 0;
}

.emotion-0 {
  color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<header
    class="emotion-3"
  >
    <div
      class="emotion-2"
    >
      <h1
        class="emotion-1"
      >
        <a
          class="emotion-0"
          href="/"
        />
      </h1>
    </div>
  </header>
</DocumentFragment>
`;

exports[`Snapshot testing Should render a header without error 1`] = `
<DocumentFragment>
  .emotion-3 {
  background: rebeccapurple;
  margin-bottom: 1.45rem;
}

.emotion-2 {
  margin: 0 auto;
  max-width: 960px;
  padding: 1.45rem 1.1rem;
}

.emotion-1 {
  margin: 0;
}

.emotion-0 {
  color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<header
    class="header emotion-3"
    data-testid="header"
  >
    <div
      class="emotion-2"
    >
      <h1
        class="emotion-1"
      >
        <a
          class="emotion-0"
          href="/"
        >
          test-title
        </a>
      </h1>
    </div>
  </header>
</DocumentFragment>
`;
